.github/workflows/jsdoc-Generate-documentation-site.yml

```diff
@@ -0,0 +1,106 @@
+# Jobs: 
+# - "generate-docs"
+#
+#   This is job uses jsdoc library to generate a static site from the comments on the library code.
+#
+#   Once the static site is generated using the command 'npm run build-jsdoc' from the 'mathtype-html-integration-devkit'
+#   package, it runs some validations to the generated files at 'packages/mathtype-html-integration-devkit/out/' folder.
+#
+#   Validation consists on:
+#    - The main index.html file is not empty.
+#    - There are no broken links on the HTML files.
+#
+#   If validation passes the static site is published as an artifact called 'mathtype-html-integration-devkit-docs.zip'. 
+
+name: Generate and validate the jsdoc site
+
+on: 
+  - push
+  - workflow_dispatch
+
+jobs:
+  generate-docs:
+    name: Generate and validate the jsdoc site
+    runs-on: ubuntu-latest    
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      # 01. install a specific version of Node using
+      # https://github.com/actions/setup-node
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+
+      # 02. Install html-integrations dependencies
+      - name: Install dependencies
+        run: |
+          npm ci
+
+      # 03. Install mathtype-html-integration-devkit dependencies
+      #     and run the jsdoc command to generate the static site.
+      - name: Install depedencies and generate the jsdoc site
+        run: |
+          cd packages/mathtype-html-integration-devkit
+          npm install
+          npm run build-jsdoc
+
+      # 04. Check if the generated documentation is empty.
+      - name: Validate main index file is not empty
+        id: check_files
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "packages/mathtype-html-integration-devkit/out/index.html"
+
+      # 05. If the main index.html file is empty, raise an error to stop
+      #     the workflow execution.
+      - name: Errors if jsdoc site is empty
+        if: steps.check_files.outputs.files_exists == 'false'
+        # Only runs if the file does not exist
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('jsdoc: main index file is empty')
```

Comment:
I believe that the element passed on this function has to be the name of the step you want to force fail. Have you checked what happens when there's no index.html with the change you added in this line?

Reply:
No, I didn't

Reply:
From the documentation, it seems that the element can have any value.
https://github.com/actions/toolkit/tree/master/packages/core